app/sagas/deepLinking.js

```diff
@@ -31,26 +31,43 @@ const handleInviteLink = function* handleInviteLink({ params, requireLogin = fal
 	}
 };
 
+const popToRoot = function popToRoot({ isMasterDetail }) {
+	if (isMasterDetail) {
+		Navigation.navigate('DrawerNavigator');
+	} else {
+		Navigation.navigate('RoomsListView');
+	}
+};
+
 const navigate = function* navigate({ params }) {
 	yield put(appStart({ root: ROOT_INSIDE }));
 	if (params.path) {
 		const [type, name] = params.path.split('/');
 		if (type !== 'invite') {
 			const room = yield RocketChat.canOpenRoom(params);
 			if (room) {
-				const isMasterDetail = yield select(state => state.app.isMasterDetail);
-				if (isMasterDetail) {
-					Navigation.navigate('DrawerNavigator');
-				} else {
-					Navigation.navigate('RoomsListView');
-				}
 				const item = {
 					name,
 					t: roomTypes[type],
 					roomUserId: RocketChat.getUidDirectMessage(room),
 					...room
 				};
-				yield goRoom({ item, isMasterDetail });
+
+				const isMasterDetail = yield select(state => state.app.isMasterDetail);
+				const focusedRooms = yield select(state => state.room.rooms);
+
+				if (focusedRooms.includes(room.rid)) {
+					// if there's one room on the list or last room is the one
+					if (focusedRooms.length === 1 || focusedRooms[0] === room.rid) {
```

Comment:
I'm curious here, if focusedRooms already includes `room.rid`, why are we checking if length equals 1? Is there a way this can have a value other than 1 in a focused room?

Reply:
`focusedRooms` is an array of all rooms opened in the app.
Try this:
- Open a group
- Tap on any avatar different than yourself
- (It's going to open RoomInfoView)
- Go to DM
- `focusedRooms` is going to have group and DM rid.